feat:es 使用 include / explode 聚合优化  --story=124743112

Zhaoyikaiii opened the pull request.

Zhaoyikaiii committed:
refactor: 改造HighlightFactory -> LabelMapFactory

- 因为Include也需要依赖这里的LabelMap作为直接改名为LabelMapFactory
- 增加FetchIncludeFieldValues
  用于提供key直接返回这个key下面的所有的values

Zhaoyikaiii committed:
refactor: 调整labelMap添加label values的条件

- 只有添加的操作符为positive才会被添加到labelMap中。
- 因为不管是include还是highlight我们都应该只会关注那些
操作符是postive的内容。
- 但是对于queryString我们并不知道每个expr之间的操作符
- 我们也就认为都是postive的。

Zhaoyikaiii committed:
refactor: 修改highlightFactory为labelMap

chenjiandongx commented on pkg/unify-query/query/structured/query_ts.go:
```diff
@@ -431,10 +436,22 @@ func (q *Query) LabelMap() (map[string][]string, error) {
 		for _, value := range condition.Value {
 			if value != "" {
-				addLabel(condition.DimensionName, value)
+				switch condition.Operator {
```
上面是有一段一模一样的代码？switch 的内容。

Zhaoyikaiii committed:
chore: 移除未使用函数

Zhaoyikaiii committed:
refactor: 增加labelMapValue记录label对应的operator

- 因为在如果operator是contains时，如果在构建include时需要包含星号(*)

Zhaoyikaiii committed:
test: 添加测试用例

Zhaoyikaiii committed:
feat(elasticsearch): add contains operator include support and refactor condition handling

- Add contains operator support for Elasticsearch aggregation include parameter
- Extract common condition operator logic to shared function package
- Add comprehensive test coverage for include functionality
- Refactor duplicate code in metadata/struct.go and query/structured/query_ts.go

Zhaoyikaiii committed:
chore: 在构建include时如果为contains操作符直接进行跳过

Zhaoyikaiii committed:
fix:Revert "chore: 在构建include时如果为contains操作符直接进行跳过"

This reverts commit a65152389cf94120e359d1e6e95dea94269e9e74.

Zhaoyikaiii committed:
refactor: 修改构建include实现

- 给formatFactory增加每个query的labelMap
- 当在进行include子句构建时直接从这个函数获取label
- 并且在构建include时只会选择操作符为eq

Zhaoyikaiii committed:
Merge branch 'master' into feat/es_agg_include_explode/#1010158081124743112

Zhaoyikaiii committed:
chore: adjust highlight.go file structure

shamcleren committed:
Merge branch 'master' into feat/es_agg_include_explode/#1010158081124743112

shamcleren committed:
feat: es 使用 include / explode 聚合优化 #1010158081124743112

shamcleren committed:
feat: es 使用 include / explode 聚合优化 #1010158081124743112

shamcleren committed:
feat: es 使用 include / explode 聚合优化 #1010158081124743112

shamcleren committed:
Merge branch 'master' into feat/es_agg_include_explode/#1010158081124743112
